lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -588,6 +641,34 @@ static void configureAIEVecV2Legalizations(ConversionTarget &target,
   });
 }
 
+static void
+configureAIEVecV1TransformationLegalizations(ConversionTarget &target) {}
+
+static void
+configureAIEVecV2TransformationLegalizations(ConversionTarget &target) {
+  target.addDynamicallyLegalOp<xilinx::aievec::BroadcastOp>(
+      [](xilinx::aievec::BroadcastOp op) {
+        if (!op.getSource().getDefiningOp())
```

Comment:
The code here (651-666) duplicates the code in FoldAIEShiftAndBroadcast::matchAndRewrite.
Is this expected?
Should we factor this out? The main difference is just the return type, but that does not prevent refactoring this.